docs/src/references/operation-workflow.md

```diff
@@ -0,0 +1,372 @@
+    Downloaded --> Failed
+    Install --> Failed
+    Installed --> Failed
+    Successful --> [*]
+    Failed --> [*]
+    
+    classDef specific color:blue;
+    class Init, Downloaded, Installed specific
+    
+    classDef plugin color:red;
+    class Download, Install plugin
+       
+    classDef mapper color:black;
+    class Successful, Failed mapper
+```
+
+### Benefits
+
+Observe on the example that:
+
+- At any state, *one and only one* participant is responsible to move the operation forward.
+- Publishing a state to the MQTT command topic, can be seen as passing the baton from one participant to another.
+  The mapper creates the `Init` state and let then the other components work.
+  The plugin tell the download has been successful by publishing the `Downloaded` state,
+  but do nothing till the domain-specific component has checked the file and move the command state to `Install`.
+- Each software component has to know only *some* states of the whole workflow:
+  - the states they are responsible for
+  - the states they create to pass the baton.
+- The compatibility of two participants, one publishing the state owned by the other, is only defined by the message payload:
+  - all the property fields expected to make progress at the new state must be provided.
+
+The benefits are that:
+- One of the participant can be substituted by another implementation as long as the substitute implementation
+  is ready to process at least all the state processed by the former implementation.
+  - This is the key principle used by thin-edge to provide extensible operation support.
+  - The `tedge-configuration-plugin` defines the `Init`, `Downloaded`, `Installed` states
+    with no specific behavior beyond proceeding to the next step;
+    so, a domain specific component can be substituted to add extra checks and actions before moving forward. 
+- Extra states and participants can be added as long as each state is owned by one of the participant.
+  - For instance, an agent developer can introduce a `Rollback` state in the `configuration-update` workflow,
+    associated by another software component responsible for these rollbacks.
+
+Furthermore, specific versions of the same workflow can be defined on different targets.
+Indeed, all the status updates for a command on a given thin-edge entity or component
+are published on an MQTT topic which prefix is the entity identifier.
+- The same executable can be used to handle operations on different targets.
+  For instance, the `tedge-configuration-plugin` can run on the main device `te/device/main//`
+  as well as on a child-device identified by `te/device/child-xyz//`.
+- A specific executable can be substituted on a specific target.
+  If for some reasons, `tedge-configuration-plugin` cannot be installed on a child-device,
+  then a specific implementation of the `configuration-update` MQTT API can be used to serve configuration updates
+  on, say, `te/micro-controller/xyz//`.
```

Comment:
> It could have been the same `te/device/child-xyz//` target, right?

Sure. I just want to give a hint of why this child device is specific: this is a micro-controller where `tedge-configuration-plugin` cannot be installed.